enhancements/olm/bundle-version-overrides.md

```diff
@@ -0,0 +1,484 @@
+
+1. Primary: Extract from CSV `spec.release` field
+2. Fallback: Parse from version build metadata (portion after `+`)
+3. Validate using semver prerelease syntax
+4. Fatal error if `olm.substitutesFor` present but build metadata invalid
+
+Build metadata processing: extract, validate, populate `release` field, clean `version` field.
+
+**Priority Comparison:**
+
+Reusable comparison methods consider both version and release fields. Release values validated as semver prerelease syntax. Empty values = lowest priority. Supports numeric, timestamp-based, or alphanumeric schemes.
+
+**Bundle Metadata:**
+
+CSV supports `spec.release` field:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: example-operator.v1.0.0
+spec:
+  version: 1.0.0
+  release: 2025.01.24.000000
+```
+
+**operator-registry Tools:**
+
+`opm` command and operator-registry library updated to:
+* Parse and extract release using hierarchical strategy
+* Validate release values using semver prerelease syntax
+* Render release fields in FBC output
+* Support explicit spec.release and build metadata extraction
+* Maintain backward compatibility
+* Add `substitutesFor` catalog template for repackaging workflows
+
+### Implementation Phases
+
+Three-phase implementation:
+
+#### Phase 1: API and Operator Registry (operator-framework/api and operator-framework/operator-registry)
+
+**Scope:**
+- Add optional `release` field to CSV specification
+- Implement release field extraction in operator-registry
+- Update `opm` tooling for parsing and rendering
+- Add `substitutesFor` catalog template
+
+**Deliverables:**
+- [ ] API schema update with `release` field in Package struct
+- [ ] Hierarchical extraction logic (spec.release → build metadata)
+- [ ] Semver prerelease validation
+- [ ] Fatal error handling for invalid build metadata with `olm.substitutesFor`
+- [ ] Reusable bundle version comparison library methods (version + release)
+- [ ] `opm render` support for release field
+- [ ] Catalog template for `substitutesFor` workflows
+- [ ] Unit and integration tests
+
+**Dependencies:** None
+
+**Status:** In Progress - See PRs:
+- https://github.com/operator-framework/api/pull/454
+- https://github.com/operator-framework/operator-registry/pull/1792
+
+#### Phase 2: OLMv1 (operator-framework/operator-controller)
+
+**Scope:**
+- Implement bundle selection prioritization by release
+- Add feature gate for release-based ordering
+- Leverage reusable bundle version comparison methods from Phase 1
+
+**Deliverables:**
+- [ ] Feature gate `ReleaseVersionPriority` (disabled by default)
+- [ ] Bundle selection logic prioritizing higher release values
+- [ ] E2E tests
+- [ ] Feature gate documentation
+
+**Dependencies:** Phase 1
+
+**Feature Gate:** `ReleaseVersionPriority` (default: disabled)
+
+**Status:** Pending
+
+#### Phase 3: OLMv0 Integration
```

Comment:
It sounds like this whole phase is just verifying that OLMv0 won't provide any kind of support for this additional field in the schema - is that accurate?

Reply:
The deliverables documented here only seem to mention that OLMv0 will just ignore all release field information here.

To clarify, this feature will _only_ be supported in OLMv1?

Reply:
Was missing the ordering requirements for v0 in phase 3. Thanks!